File: rentermatch/src/main/java/com/humanlearning/rentermatch/domain/Tenant.java
package com.humanlearning.rentermatch.domain;

import java.util.List;
import lombok.Data;

@Data
public class Tenant {

  private String tid;
  private Integer tAge;
  private String tClientId;
  private String tConstellation;
  private String tCooking;
  private String tEarlyTimeSleep;
  private Integer tExpenditure;
  private String tGender;
  private String tJob;
  private String tLateTimeSleep;
  private List<String> tMatches;
  private Integer tNumOfRoomates;
  private String tPet;
  private String tPhone;
  private String tPreferLocation;
  private String tPreferType;
  private String tPreferZipCode;
  private String tSmoking;
}

File: rentermatch/src/main/resources/static/tenant.ts
import {client} from "./client";

export class tenant extends client {

  constructor(
      public username: string,
      public password: string,
      public Email: string,
      public tPhone: string,
      public tGender: string,
      public tAge: number,
      public tConstellation: string,
      public tJob: string,
      public tPreferLocation: string,
      public tPreferZipCode: number,
      public tPreferType: string,
      public tExpenditure: number,
      public tNumOfRoomates: number,
      public tSmoking: string,
      public tPet: string,
      public tCooking: string,
      public tEarlyTimeSleep: string,
      public tLateTimeSleep: string,
      public uni: string
  ) {
    super(username, password, Email);
  }

}

export interface tenantRsp {
  data: tenant
}
